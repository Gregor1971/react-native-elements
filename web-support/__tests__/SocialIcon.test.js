import React from 'react';
import {
  Text,
  TouchableHighlight,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import parseSamples, {
  assignRef,
  genRefId,
} from 'enzyme-styleguidist-sample-parser';
import options, {
  snapShot,
  buildJsxForGuideMethod,
  ensureCalled,
  onlyEnsureCalled,
  onlySnapshots,
} from '../samples';
import { SocialIcon as Component } from '../../src';

const props = {
  noProps: {
    component: Component,
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  title: {
    component: Component,
    props: { type: 'medium', title: 'medium', button: true },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  type: {
    component: Component,
    props: { type: 'medium' },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  raised: {
    component: Component,
    props: { type: 'medium', raised: false },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  button: {
    component: Component,
    props: { type: 'medium', button: true },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  onPress: {
    component: Component,
    props: {
      type: 'medium',
      onPress: () => {
        console.log('pressed');
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  onLongPress: {
    component: Component,
    props: {
      type: 'medium',
      onLongPress: () => {
        console.log('pressed, long');
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  light: {
    component: Component,
    props: { type: 'medium', light: true },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  iconStyle: {
    component: Component,
    props: {
      type: 'medium',
      button: true,
      iconStyle: { transform: [{ rotateZ: '-45deg' }] },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  style: {
    component: Component,
    props: {
      type: 'medium',
      button: true,
      style: { width: 120, transform: [{ rotateZ: '-45deg' }] },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  iconColor: {
    component: Component,
    props: { type: 'medium', iconColor: '#071' },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  underlayColor: {
    component: Component,
    props: {
      type: 'medium',
      onPress: () => {},
      underlayColor: 'yellow',
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  iconSize: {
    component: Component,
    props: { type: 'medium', iconSize: 40 },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  component: {
    component: Component,
    props: { type: 'medium', component: TouchableWithoutFeedback },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      getJsxString: () => {
        return `const TouchableWithoutFeedback = RN.TouchableWithoutFeedback;
  <SocialIcon
    type='medium'
    component={TouchableWithoutFeedback}
  />`;
      },
    },
  },
  fontFamily: {
    component: Component,
    props: {
      type: 'medium',
      fontFamily: 'Courier New',
      title: 'medium',
      button: true,
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  fontWeight: {
    component: Component,
    props: {
      type: 'medium',
      fontWeight: 'bold',
      title: 'medium',
      button: true,
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  fontStyle: {
    component: Component,
    props: {
      type: 'medium',
      fontStyle: { padding: 50 },
      title: 'medium',
      button: true,
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  disabled: {
    component: Component,
    props: { type: 'medium', disabled: true },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  loading: {
    component: Component,
    props: { type: 'medium', loading: true, title: 'medium', button: true },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
};

const samples = {
  props,
};

parseSamples(
  {
    Icons: { sectionComponents: { SocialIcon: { samples } } },
  },
  options
);
