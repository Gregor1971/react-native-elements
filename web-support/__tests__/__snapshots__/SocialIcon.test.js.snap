// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons: SocialIcon: [ prop: button ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      96,
      false,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: component ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: disabled ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={true}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: fontFamily ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      96,
      false,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
    <TextElement
      style={
        Array [
          99,
          undefined,
          Object {
            "fontFamily": "Courier New",
          },
          undefined,
          undefined,
        ]
      }
    >
      medium
    </TextElement>
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: fontStyle ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      96,
      false,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
    <TextElement
      style={
        Array [
          99,
          undefined,
          undefined,
          undefined,
          Object {
            "padding": 50,
          },
        ]
      }
    >
      medium
    </TextElement>
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: fontWeight ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      96,
      false,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
    <TextElement
      style={
        Array [
          99,
          undefined,
          undefined,
          Object {
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      medium
    </TextElement>
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: iconColor ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="#071"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: iconSize ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={40}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: iconStyle ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      96,
      false,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          Object {
            "transform": Array [
              Object {
                "rotateZ": "-45deg",
              },
            ],
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: light ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      Object {
        "backgroundColor": "white",
      },
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="#02b875"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: loading ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      96,
      false,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
    <TextElement
      style={
        Array [
          99,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      medium
    </TextElement>
    <ActivityIndicator
      animating={true}
      color="white"
      hidesWhenStopped={true}
      size="large"
      style={
        Array [
          101,
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: noProps ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": undefined,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: onLongPress ] enzyme-shallow: snapshot 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  disabled={false}
  onLongPress={[Function]}
  onPress={[Function]}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
  underlayColor="black"
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</TouchableHighlight>
`;

exports[`Icons: SocialIcon: [ prop: onPress ] enzyme-shallow: snapshot 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  disabled={false}
  onLongPress={[Function]}
  onPress={[Function]}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
  underlayColor="black"
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</TouchableHighlight>
`;

exports[`Icons: SocialIcon: [ prop: raised ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      false,
      95,
      false,
      false,
      Object {
        "borderRadius": 48,
        "height": 52,
        "width": 52,
      },
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: style ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      96,
      false,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      Object {
        "transform": Array [
          Object {
            "rotateZ": "-45deg",
          },
        ],
        "width": 120,
      },
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: title ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      96,
      false,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
    <TextElement
      style={
        Array [
          99,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      medium
    </TextElement>
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: type ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: underlayColor ] enzyme-shallow: snapshot 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  disabled={false}
  onLongPress={[Function]}
  onPress={[Function]}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
  underlayColor="black"
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</TouchableHighlight>
`;
